wbobeirne/joule-extension
# Sync migrations

Closes #137

### Description

Adds a `version: number` configuration to each synced data store that allows you to increment it and provide a migration dict entry to migrate the data.

This was intended to fix the rates migration needed as outlined in #137. But after testing, I realized that the rates cache _didn't actually work_. Rates were being fetched and applied to the store more often than not _before_ the rates cache was being set, which means that ironically the rates cache would more often than not override your fresh rates. While I could sort through this one and get it working, I decided to chop the gordian knot here and cut the rates cache.

However, the migration stuff worked, so I decided to PR it in anyway. A guide for how to write migrations has been added here: https://github.com/wbobeirne/joule-extension/wiki/Synced-data-migrations

### Steps to Test

1. Open Joule and make sure it still works.
2. (If you're so inclined) write a migration just to give it a shot.
